src/main/java/com/lunark/lunark/reservations/service/ReservationService.java

```diff
@@ -89,6 +89,99 @@ public List<Reservation> getAllReservationsForPropertiesList(List<Property> prop
+        }
+        return reservationsForProperties;
+    }
+
+    @Override
+    public List<Reservation> getAllAcceptedReservations(Long guestId) {
+        return reservationRepository.findAll().stream().filter(reservation -> Objects.equals(reservation.getGuest().getId(), guestId) && reservation.getStatus() == ReservationStatus.ACCEPTED).toList();
+    }
+
+    @Override
+    public void save(Reservation reservation) {
+        Optional<Reservation> reservationUpdate = findById(reservation.getId());
+        if (reservationUpdate.isPresent()) {
+            Reservation existingReservation = reservationUpdate.get();
+            existingReservation.copyFields(reservation);
+            reservationRepository.saveAndFlush(existingReservation);
+        } else {
+            throw new RuntimeException("Reservation not found with id: " + reservation.getId());
+        }
+    }
+
+    @Override
+    public void updateReservations(Reservation reservation) {
+        Long propertyId = reservation.getProperty().getId();
+        List<Reservation> allPropertyReservations = reservationRepository.findByPropertyId(propertyId);
+
+        for (Reservation existingReservation : allPropertyReservations) {
+            if(existingReservation.getId().equals(reservation.getId()))
+                continue;
+            if (doDatesOverlap(reservation, existingReservation)) {
+                existingReservation.setStatus(ReservationStatus.REJECTED);
+                reservationRepository.save(existingReservation);
+            }
+        }
+    }
+
+    private boolean doDatesOverlap(Reservation newReservation, Reservation existingReservation) {
+        return newReservation.getStartDate().isBefore(existingReservation.getEndDate()) &&
+                existingReservation.getStartDate().isBefore(newReservation.getEndDate()) ||
+                newReservation.getStartDate().isEqual(existingReservation.getEndDate()) ||
+                newReservation.getEndDate().isEqual(existingReservation.getStartDate());
+    }
+
+    @Override
+    public void acceptOrRejectReservation(Reservation reservation, ReservationStatus isAccepted) {
+        reservation.setStatus(isAccepted);
+        if(isAccepted == ReservationStatus.ACCEPTED) {
+            updatePropertyAvailability(reservation, true);
+            updateReservations(reservation);
+        }
+        save(reservation);
+    }
+
+    @Override
+    public boolean cancelReservation(Reservation reservation) {
+        if (isPastCancellationDeadline(reservation)) {
+            return false;
+        }
+        reservation.setStatus(ReservationStatus.CANCELLED);
+        updatePropertyAvailability(reservation, false);
+        save(reservation);
+        return true;
+    }
+
+    private boolean isPastCancellationDeadline(Reservation reservation) {
+        Property property = reservation.getProperty();
+        LocalDateTime currentDateTime = LocalDateTime.now();
```

Comment:
Inject a `Clock` into this class so this method can be tested.
```suggestion
        LocalDateTime currentDateTime = LocalDateTime.now(clock);
```